### PR TITLE
Embl predoc2017 notebooks

### PR DESCRIPTION
**Pull request by eleanorwilliams**
This is the latest version of the notebooks ready for final checks. Both notebooks for the training course are in this PR

### CODE REVIEW

**Comment by eleanorwilliams** [comment]
@manics Did you say yesterday that there was something I should merge into this branch? PR13?
@bramalingam Unique list of genes now obtained from the Phenotype to Genes query.

**Comment by bramalingam** [comment]
@eleanorwilliams : changes look good. Could we fix the small error in https://github.com/IDR/jupyter-docker/pull/19#issuecomment-344912607 and hand it over to @manics for deployment. We can do a final check of the notebooks once the deployment is done.

**Comment by eleanorwilliams** [comment]
@bramalingam  I've made a change so that the phenotype widget is automatically populated with 'elongated cell phenotype'.  This is different behaviour to the widget above for genes, where it gives the suggestion of 'Enter GO Id e.g. GO:0005885' and then gives a message when empty, but I chose to do this way because the suggestion of 'Enter phenotype e.g. elongated cell phenotype' is too long for the box and I couldn't figure out how to make the box wider.  

Three RFEs would be:

1.  Make text box longer and then be consistent in our way of giving suggestions for values
2.  Write a check to make sure the value entered is a valid Phenotype Term Name  before going to the next step of querying for genes. 
3. Allow CMPO accessions to be entered.

File: notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+    "gene_list = list(gene_set) \n",
+    "print gene_list\n",
+    "print \"\\nThere are\", len(gene_list), \"unique genes\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Now we can perform enrichment analysis on the the gene set associated with the phenotype \n",
```
Comment (jburel) [inline notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb]: double ``the the`` before  ``gene``

File: notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    "---\n",
+    "## Phenotypes to genes query"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Choose a phenotype that we'd like to query\n",
+    "Select a Phenotype Term Name E.g. elongated cell phenotype"
```
Comment (jburel) [inline notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb]: ``e.g.``

File: notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+    "print \"\\nThere are\", len(gene_list), \"unique genes\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Now we can perform enrichment analysis on the the gene set associated with the phenotype \n",
+    "It will find which terms are over-represented using annotations for that gene set.\n",
+    "We are going to use the python package called GSEApy to do this.  It performs gene set enrichment analysis by calling the enrichr online api. See  http://pythonhosted.org/gseapy/run.html for more information about the package, and http://amp.pharm.mssm.edu/Enrichr/ for more information about Enrichr."
```
Comment (jburel) [inline notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb]: Python

File: notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb (hunk 2 above)
Comment (eleanorwilliams) [reply notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb]: fixed

File: notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb (hunk 1 above)
Comment (eleanorwilliams) [reply notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb]: fixed

File: notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb (hunk 3 above)
Comment (eleanorwilliams) [reply notebooks/EMBLPreDocTraining/GeneAndPhenotypeQueries.ipynb]: fixed

**Comment by sbesson** [comment]
Consumed and used in the context of the EMBL 2017 training. We probably do not want to get this merged into IDR under this form although we need to sort out the layout for these training notebooks for the foreseeable future. Once we have a structure into place, these changes can be integrated to document the state of the training. Closing for now.

**Pull request closed without merging**